timemachine/fe/free_energy.py

```diff
@@ -51,19 +51,31 @@ class InitialState:
 
 
 @dataclass
-class SimulationResult:
+class PairBarResult:
     all_dGs: List[float]  # L - 1
     all_errs: List[float]  # L - 1
-    dG_errs_by_lambda_by_component: np.ndarray  # (len(U_names), L - 1)
+    dG_errs_by_lambda_by_component: NDArray  # (len(U_names), L - 1)
     overlaps_by_lambda: List[float]  # L - 1
-    overlaps_by_lambda_by_component: np.ndarray  # (len(U_names), L - 1)
+    overlaps_by_lambda_by_component: NDArray  # (len(U_names), L - 1)
+    u_kln_by_lambda_by_component: NDArray
+
+
+@dataclass
+class PairBarPlots:
     dG_errs_png: bytes
     overlap_summary_png: bytes
     overlap_detail_png: bytes
-    frames: List[Union[np.ndarray, StoredArrays]]  # (len(keep_idxs), n_frames, N, 3)
-    boxes: List[np.ndarray]
+
+
+@dataclass
+class SimulationResult:
     initial_states: List[InitialState]
+    result: PairBarResult
+    plots: PairBarPlots
+    frames: List[NDArray]  # (len(keep_idxs), n_frames, N, 3)
+    boxes: List[NDArray]
     md_params: MDParams
+    intermediate_results: List[Tuple[List[InitialState], PairBarResult]]
```

Comment:
I was on the fence about adding this field in this PR, since it's redundant for the existing sequential method, but serves to illustrate the motivation for adding `PairBarResult`.